pkg/common/telemetry/metrics.go

```diff
@@ -83,12 +84,19 @@ func NewMetrics(c *MetricsConfig) (*MetricsImpl, error) {
 		} else {
 			conf.EnableHostnameLabel = true
 		}
+
 		conf.EnableTypePrefix = runner.requiresTypePrefix()
 		conf.AllowedLabels = c.FileConfig.AllowedLabels
 		conf.BlockedLabels = c.FileConfig.BlockedLabels
 		conf.AllowedPrefixes = c.FileConfig.AllowedPrefixes
 		conf.BlockedPrefixes = c.FileConfig.BlockedPrefixes
 
+		if c.FileConfig.EnableTrustDomainLabel != nil {
+			impl.enableTrustDomainLabel = *c.FileConfig.EnableTrustDomainLabel
+		} else {
+			impl.enableTrustDomainLabel = false
+		}
```

Comment:
since default is false you can just avoid else, or set 
```
impl.enableTrustDomainLabel = false
if c.FileConfig.EnableTrustDomainLabel != nil {
	impl.enableTrustDomainLabel = *c.FileConfig.EnableTrustDomainLabel
} 
```